src/main/java/org/javacs/JavaLanguageServer.java

```diff
@@ -641,15 +642,17 @@ private void updateActiveFile(URI uri) {
         var el = activeFileCache.element(line, column);
         if (!el.isPresent()) return Optional.empty();
 
-        // Add code hover message
         var result = new ArrayList<MarkedString>();
-        var code = hoverCode(el.get());
-        result.add(new MarkedString("java.hover", code));
+
         // Add docs hover message
         var docs = hoverDocs(el.get());
-        if (docs.isPresent()) {
-            result.add(new MarkedString("markdown", docs.get()));
-        }
+        docs.filter(Predicate.not(String::isBlank)).ifPresent(doc -> {
+            result.add(new MarkedString(doc));
+        });
+
+        // Add code hover message
+        var code = hoverCode(el.get());
+        result.add(new MarkedString("java", code));
```

Comment:
If #85 is not something that should be merged, I can re-add the `.hover` here.

Reply:
I commented on #85 why java.hover is intentional

Reply:
Got it 👍 . I see you added it back in bfd7016886beb986e588bddc818b35af2fd2c16b.